internal/resources/diffValidation/resource_nsxt_lb_profile.go

```diff
@@ -0,0 +1,140 @@
+//  (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package diffvalidation
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	profiles        = "profile_type"
+	httpProfile     = "http_profile"
+	tcpProfile      = "tcp_profile"
+	udpProfile      = "udp_profile"
+	cookieProfile   = "cookie_profile"
+	sourceIpProfile = "sourceip_profile"
+	genericProfile  = "generic_profile"
+	clientProfile   = "client_profile"
+	serverProfile   = "server_profile"
+
+	LBHttpProfile                = "LBHttpProfile"
+	LBFastTcpProfile             = "LBFastTcpProfile"
+	LBFastUdpProfile             = "LBFastUdpProfile"
+	LBCookiePersistenceProfile   = "LBCookiePersistenceProfile"
+	LBSourceIpPersistenceProfile = "LBSourceIpPersistenceProfile"
+	LBGenericPersistenceProfile  = "LBGenericPersistenceProfile"
+	LBClientSslProfile           = "LBClientSslProfile"
+	LBServerSslProfile           = "LBServerSslProfile"
+
+	applicationProfile = "application-profile"
+	persistenceProfile = "persistence-profile"
+	sslProfile         = "ssl-profile"
+)
+
+type LoadBalancerProfile struct {
+	diff *schema.ResourceDiff
+}
+
+func NewLoadBalancerProfileValidate(diff *schema.ResourceDiff) *LoadBalancerProfile {
+	return &LoadBalancerProfile{
+		diff: diff,
+	}
+}
+
+func (l *LoadBalancerProfile) DiffValidate() error {
+	err := l.validateProfile()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (l *LoadBalancerProfile) validateProfile() error {
+	pTypes := l.diff.Get(profiles)
+	switch pTypes {
+	case applicationProfile:
+		p := l.diff.Get(httpProfile)
+		if len((p).([]interface{})) != 0 {
+			for _, profile := range p.([]interface{}) {
+				profile_type := profile.(map[string]interface{})["service_type"].(string)
+				if profile_type != LBHttpProfile {
+					return fmt.Errorf("please provide service_type as " + LBHttpProfile + " for " + httpProfile + " configuration")
+				}
+			}
+		}
```

Comment:
Cant these lines 58-66 be made a function and reused everywhere below ?

Reply:
no we can not do it as separate because each application having 3 differet service type like http, udp,tcp so not able to make it generic function

Reply:
updated

Reply:
I think u haven't addressed my comment.